src/instructlab/configuration.py

```diff
@@ -498,48 +632,104 @@ def fill_defaults(
 class _train(BaseModel):
     # model configuration
     model_config = ConfigDict(extra="ignore", protected_namespaces=())
-
-    model_path: str
-
-    data_path: str
-    ckpt_output_dir: str
-    data_output_dir: str
-
-    max_seq_len: int
-    max_batch_len: int
-    num_epochs: int
-    effective_batch_size: int
-    save_samples: int
-    checkpoint_at_epoch: bool = True
-
-    deepspeed_cpu_offload_optimizer: bool
-
-    lora_rank: int | None = None
-    lora_quantize_dtype: str | None = None
-
-    is_padding_free: bool
-
-    nproc_per_node: int
-
-    additional_args: dict[str, typing.Any]
-
+    model_path: str = Field(
+        default=DEFAULTS.MODEL_REPO,
+        description="Directory where the model to be trained is stored.",
+    )
+    data_path: str = Field(
+        default=DEFAULTS.DATASETS_DIR,
+        description="Directory where datasets used for training are stored.",
+    )
+    ckpt_output_dir: str = Field(
+        default=DEFAULTS.CHECKPOINTS_DIR,
+        description="Directory where periodic training checkpoints are stored.",
+    )
+    data_output_dir: str = Field(
+        default=DEFAULTS.INTERNAL_DIR,
+        description="Directory where the processed training data is stored (post filtering/tokenization/masking).",
+    )
+    max_seq_len: int = Field(
+        default=4096,
+        description="Maximum sequence length to be included in the training set. Samples exceeding this length will be dropped.",
+    )
+    max_batch_len: int = Field(
+        default=10000,
+        description="Maximum tokens per gpu for each batch that will be handled in a single step. If running into out-of-memory errors, this value can be lowered but not below the `max_seq_len`.",
+    )
+    num_epochs: int = Field(
+        default=10, description="Number of epochs to run training for."
+    )
+    effective_batch_size: int = Field(
+        default=3840,
+        description="The number of samples in a batch that the model should see before its parameters are updated.",
+    )
+    save_samples: int = Field(
+        default=250000,
+        description="Number of samples the model should see before saving a checkpoint.",
+    )
+    checkpoint_at_epoch: bool = Field(
+        default=True, description="Save a checkpoint at the end of each epoch."
+    )
+    deepspeed_cpu_offload_optimizer: bool = Field(
+        default=False, description="Allow CPU offload for deepspeed optimizer."
+    )
+    lora_rank: int | None = Field(
+        default=None,
+        description="Rank of low rank matrices to be used during training.",
+    )
+    lora_quantize_dtype: str | None = Field(
+        default=None,
+        description="The data type for quantization in LoRA training. Valid options are 'None' and 'nf4'.",
```

Comment:
Yes but I just recopied what was there before. In this case, None will translate to empty it seems.

Reply:
This will be visible in config file. Users will use it as a reference. If I put "None" there as a value, is it interpreted as "None" string or as an empty value?

Reply:
@RobotSail do you know? Thanks. Where is `lora_quantize_dtype` used?

Reply:
The valid values are `"nf4"` and `None`.

Reply:
`None` as in string literal "None", or as in Python value `None`? Basically, what happens when the user puts "None" string there? Is it accepted as a `None` (null) value, or is it considered an actual string?